src/types/nbodysystem.rs

```diff
@@ -98,4 +123,106 @@ impl NBodySystem {
                 },
             )
     }
+
+    pub fn set_output(&mut self, output: String) {
+        self.m_output = output;
+    }
+
+    pub fn set_limit(&mut self, limit: u16) {
+        self.m_limit = limit;
+    }
+
+    pub fn set_file(&mut self, file: bool) {
+        self.m_file = file;
+    }
+
+    fn output(&self) {
+        static OUTPUT: AtomicU16 = AtomicU16::new(1);
+
+        let count = OUTPUT.fetch_add(1, Ordering::SeqCst);
```

Comment:
Technically it's possible to use there slightly weaker memory model(Acquire/Release semantics should be fine). It's a minor thing. Also, it's better to consider, whether this part should be protected at all by atomic or a simple mutex can be used instead.